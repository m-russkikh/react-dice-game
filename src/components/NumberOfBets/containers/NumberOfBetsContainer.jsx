import React, { Component } from 'react';

import InputControl from '../../InputControl';
import { NumberOfBetsContainerPropType } from '../propTypes';


class NumberOfBetsContainer extends Component {
  handleChange = ({ currentTarget: { value } }) => {
    this.props.setNumberOfBets(parseInt(value, 10));
  }

  render() {
    return (
      <InputControl
        label="Number of Bets"
        onChange={this.handleChange}
      />
    );
  }
}

NumberOfBetsContainer.propTypes = NumberOfBetsContainerPropType;

export default NumberOfBetsContainer;
