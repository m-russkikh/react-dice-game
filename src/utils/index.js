import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
