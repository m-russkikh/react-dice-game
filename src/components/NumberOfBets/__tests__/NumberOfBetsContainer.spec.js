import { getShallowWrapper } from '../../../utils/testUtils';
import NumberOfBetsContainer from '../containers/NumberOfBetsContainer';


describe('NumberOfBetsContainer', () => {
  let props;
  let wrapper;
  let instance;

  beforeEach(() => {
    props = {
      setNumberOfBets: jest.fn(),
    };

    wrapper = getShallowWrapper(NumberOfBetsContainer, props);
    instance = wrapper.instance();
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('handleChange', () => {
      it('should call setNumberOfBets', () => {
        const event = { currentTarget: { value: 5 } };

        instance.handleChange(event);

        expect(props.setNumberOfBets).toHaveBeenCalledTimes(1);
        expect(props.setNumberOfBets).toHaveBeenCalledWith(event.currentTarget.value);
      });
    });
  });
});
