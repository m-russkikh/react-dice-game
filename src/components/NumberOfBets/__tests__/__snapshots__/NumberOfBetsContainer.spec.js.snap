// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberOfBetsContainer render should display correct 1`] = `"<InputControlView label=\\"Number of Bets\\" onChange={[Function]} />"`;
